library/Icinga/Data/UserSuggestions.php

```diff
@@ -0,0 +1,154 @@
+     * Get User group backend
+     *
+     * @return mixed $userGroupBackend
+     */
+    public function getUserGroupBackend()
+    {
+        return $this->userGroupBackend;
+    }
+
+    /**
+     * Set User group name
+     *
+     * @param string $userGroupName
+     */
+    public function setUserGroupName(string $userGroupName): self
+    {
+        $this->userGroupName = $userGroupName;
+
+        return $this;
+    }
+
+    /**
+     * Get User group name
+     *
+     * @return string $userGroupName
+     */
+    public function getUserGroupName(): string
+    {
+        return $this->userGroupName;
+    }
+
+    /**
+     * Get backends
+     *
+     * @return mixed
+     */
+    public function getBackends()
+    {
+        return $this->backends;
+    }
+
+    /**
+     * Set backends
+     *
+     * @param mixed $backends
+     */
+    public function setBackends($backends): self
+    {
+        $this->backends = $backends;
+
+        return $this;
+    }
+
+    protected function fetchSuggestions(string $searchTerm, array $exclude)
+    {
+        $count = 0;
+        foreach ($this->getBackends() as $backend) {
+            try {
+                if ($backend instanceof DomainAwareInterface) {
+                    $domain = $backend->getDomain();
+                } else {
+                    $domain = null;
+                }
+
+                $members = $this->getUserGroupBackend()
+                    ->select()
+                    ->from('group_membership', ['user_name'])
+                    ->where('group_name', $this->getUserGroupName())
+                    ->fetchColumn();
+
+
+                if (! empty($exclude)) {
+                    $members = array_merge($members, $exclude);
+                }
+
+                $filter = Filter::matchAll(
+                    Filter::where('user_name', $searchTerm),
+                    Filter::not(Filter::where('user_name', $members))
+                );
+
+                $users = $backend->select(['user_name'])
+                    ->limit(self::DEFAULT_LIMIT)
+                    ->applyFilter($filter)
+                    ->fetchColumn();
+
+                foreach ($users as $userName) {
+                    if ($count === self::DEFAULT_LIMIT) {
+                        return;
+                    }
+
+                    $userObj = new User($userName);
+                    if ($domain !== null) {
+                        if ($userObj->hasDomain() && $userObj->getDomain() !== $domain) {
+                            // Users listed in a user backend which is configured to be responsible for a domain should
+                            // not have a domain in their username. Ultimately, if the username has a domain, it must
+                            // not differ from the backend's domain. We could log here - but hey, who cares :)
+                            continue;
+                        } else {
+                            $userObj->setDomain($domain);
```

Comment:
Doesn't this set a domain for users without any? Why?

Reply:
It sets the domain, if $domain !== null.